src/tr2/game/ui/dialogs/graphic_settings.c

```diff
@@ -186,6 +388,16 @@ static bool M_RequestChangeValue(const int32_t row_idx, const int32_t dir)
     case COT_INT32:
         *(int32_t *)option->target += delta;
         break;
+    case COT_DOUBLE: {
+        const float df = ceilf((delta / 100.0f) * 10) / 10;
+        *(double *)option->target += df;
+        break;
+    }
+    case COT_FLOAT: {
+        const float df = ceilf((delta / 100.0f) * 10) / 10;
+        *(float *)option->target += df;
+        break;
```

Comment:
I think even with this approach, it can accumulate error over time. Maybe we could do something in the lines of
```c
*(float*)option->target = (round(*(float*)option->target * 10) + delta) / 10.0f;
```
?

Reply:
Thanks, updated. We still need to divide the delta alone as it's an int. I suppose the alternative could be having `delta_slow`/`delta_fast` as floats themselves.